### PR TITLE
Add /stats command with video poster and reaction tracking

### PR DESCRIPTION
**Pull request by jaantaponen**
## Summary

- Tracks who posts videos via `/dl` and records stats (poster, URL, timestamp, repost flag) in DuckDB after each successful send
- Tracks 👍/👎 reactions per video on both Telegram (via `MiddlewarePoller` diffing `OldReaction`/`NewReaction` arrays) and Discord (via `MessageReactionAdd`/`Remove` events)
- New `/stats` command shows a per-group leaderboard: top posters, most liked, most disliked, and top reposters
- Schema migration is automatic — existing DBs get the new columns added via `ALTER TABLE IF NOT EXISTS` on first run

## Test plan

- [ ] `/dl <url>` in a group → row appears in `video_stats` table
- [ ] React 👍 to the bot's video message → `thumbs_up_count` increments
- [ ] React 👎 → `thumbs_down_count` increments
- [ ] Change reaction from 👍 to 👎 → counts update correctly (net zero on 👍, +1 on 👎)
- [ ] `/stats` returns leaderboard with all four sections
- [ ] Repost a video → sender appears in "Top reposters" section
- [ ] Existing DB without new columns migrates without data loss

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/utils/stats_db.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+}
+
+func queryTopByReactionColumn(dbPath, groupId, column string, limit int) ([]ReactionStat, error) {
+	conn, err := sql.Open("duckdb", dbPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open DuckDB: %w", err)
+	}
+	defer conn.Close()
+
+	query := fmt.Sprintf(`
```
Comment (Napuu) [inline pkg/utils/stats_db.go:139]: **SQL injection risk (latent):** `column` is interpolated directly into the SQL via `fmt.Sprintf`. Currently only called with hardcoded strings so it's safe today, but this is a fragile pattern. Consider adding an allowlist check:
```go
if column != "thumbs_up_count" && column != "thumbs_down_count" {
    return nil, fmt.Errorf("invalid column: %s", column)
}
```

File: internal/handlers/stats_handler.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			platform = "telegram"
+		case Discord:
+			platform = "discord"
+		}
+		groupId := platform + ":" + m.chatId
+
+		posters, postersErr := utils.GetGroupLeaderboard(dbPath, groupId, 10)
+		thumbsUp, thumbsUpErr := utils.GetTopThumbsUp(dbPath, groupId, 5)
+		thumbsDown, thumbsDownErr := utils.GetTopThumbsDown(dbPath, groupId, 5)
+		reposters, repostersErr := utils.GetTopReposters(dbPath, groupId, 5)
```
Comment (Napuu) [inline internal/handlers/stats_handler.go]: **DB connection churn:** Each of these 4 calls opens and closes a separate DuckDB connection. Consider opening one `*sql.DB` connection here and passing it to all 4 queries. Same applies to the reaction tracking handlers — every reaction event opens/closes a connection, which could become a bottleneck on busy servers.

File: pkg/utils/repost_db.go [hunk 3]
```diff
@@ -40,7 +40,27 @@ func InitRepostDB(dbPath string) error {
...
+			bot_message_id TEXT NOT NULL,
+			reaction_count  INT NOT NULL DEFAULT 0,
+			thumbs_up_count INT NOT NULL DEFAULT 0,
+			thumbs_down_count INT NOT NULL DEFAULT 0,
+			is_repost     BOOLEAN NOT NULL DEFAULT FALSE,
+			posted_at     TIMESTAMP NOT NULL
+		);
+		CREATE UNIQUE INDEX IF NOT EXISTS idx_video_stats_lookup
+			ON video_stats (platform, group_id, bot_message_id);
+		ALTER TABLE video_stats ADD COLUMN IF NOT EXISTS thumbs_up_count INT DEFAULT 0;
```
Comment (Napuu) [inline pkg/utils/repost_db.go:63]: **Redundant migration:** These `ALTER TABLE ... ADD COLUMN IF NOT EXISTS` statements add columns already defined in the `CREATE TABLE IF NOT EXISTS` above. For new DBs both execute (works but is confusing). For existing DBs only the `ALTER TABLE` matters. Consider either removing the columns from `CREATE TABLE` (relying solely on `ALTER TABLE` for a clearer migration path) or adding a comment explaining this is intentional for migrating existing DBs.

File: pkg/utils/stats_db.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+func UpdateReactionCount(dbPath string, platform, groupId, botMessageId, emoji string, delta int) error {
+	conn, err := sql.Open("duckdb", dbPath)
+	if err != nil {
+		return fmt.Errorf("failed to open DuckDB: %w", err)
+	}
+	defer conn.Close()
+
+	query := `
+		UPDATE video_stats
+		SET reaction_count    = reaction_count + ?,
```
Comment (Napuu) [inline pkg/utils/stats_db.go]: **Unused column:** `reaction_count` gets incremented on every reaction, but no query ever reads it — the leaderboard only uses `thumbs_up_count` and `thumbs_down_count`. Either remove it or use it for something (e.g., a "most reacted" leaderboard).

File: pkg/utils/stats_db.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to open DuckDB: %w", err)
+	}
+	defer conn.Close()
+
+	query := `
+		SELECT user_id, username, COUNT(*) AS post_count
+		FROM video_stats
+		WHERE group_id = ? AND is_repost = FALSE
+		GROUP BY user_id, username
```
Comment (Napuu) [inline pkg/utils/stats_db.go]: **Username changes break grouping:** If a user changes their username, they'll appear as two separate entries in the leaderboard. Consider grouping only by `user_id` and using `MAX(username)` or a subquery for the latest name.

File: internal/handlers/video_stats_handler.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+func (h *VideoStatsHandler) Execute(m *Context) {
+	slog.Debug("Entering VideoStatsHandler")
+
+	if m.sendVideoSucceeded && m.action == DownloadVideo && m.botMessageId != "" {
+		cfg := config.FromEnv()
+		dbPath := filepath.Join(cfg.REPOST_DB_DIR, statsDBFileName)
+
+		var platform string
+		switch m.Service {
```
Comment (Napuu) [inline internal/handlers/video_stats_handler.go:26]: **Missing `default` case:** If `platform` stays empty (e.g., a new service is added), the `groupId` would be `:chatId` which could cause data integrity issues. Consider adding a `default` that logs a warning and returns early.

**Review by Napuu — COMMENTED** [review]
Good feature overall — clean chain integration, solid test coverage, and nice handling of edge cases (empty states, singular/plural). A few suggestions below, mostly around DB connection handling and future-proofing.

File: pkg/utils/repost_db.go (hunk 3 above)
Comment (jaantaponen) [reply pkg/utils/repost_db.go:63]: Addressed in 91e0333 — kept the `ALTER TABLE` block (needed for pre-existing DBs) and added a comment explaining why it duplicates the `CREATE TABLE` columns and why the defaults are looser (DuckDB doesn't allow NOT NULL with ADD COLUMN).

File: pkg/utils/stats_db.go (hunk 1 above)
Comment (jaantaponen) [reply pkg/utils/stats_db.go:139]: Fixed in 91e0333 — added allowlist guard that rejects anything other than `thumbs_up_count`/`thumbs_down_count`.

File: pkg/utils/stats_db.go (hunk 5 above)
Comment (jaantaponen) [reply pkg/utils/stats_db.go]: Fixed in 91e0333 — `GetGroupLeaderboard` and `GetTopReposters` now `GROUP BY user_id` with `MAX(username)` for display. Added a regression test (`TestGetGroupLeaderboardGroupsByUserId`).

File: internal/handlers/video_stats_handler.go (hunk 6 above)
Comment (jaantaponen) [reply internal/handlers/video_stats_handler.go:26]: Fixed in 91e0333 — added a `default` case to the platform switches in both `VideoStatsHandler` and `StatsHandler` that logs a warning and returns early before touching the DB.

File: internal/handlers/stats_handler.go (hunk 2 above)
Comment (jaantaponen) [reply internal/handlers/stats_handler.go]: Fixed in 91e0333 — `StatsHandler` now opens one `*sql.DB` via `OpenStatsDB` and passes it to all 4 queries. Discord and Telegram reaction tracking also share a single persistent connection opened at bot startup instead of per-event.

File: pkg/utils/stats_db.go (hunk 4 above)
Comment (jaantaponen) [reply pkg/utils/stats_db.go]: Fixed in 91e0333 — removed `reaction_count` from the schema and from `UpdateReactionCount`. The per-emoji counts were already covering every read site.
